src/test/java/io/vertx/core/spi/executor/ExecutorFactoryMetricsTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.spi.executor;
+
+import static io.vertx.core.spi.executor.Utils.EXECUTE;
+import static io.vertx.core.spi.executor.Utils.SHUTDOWN_NOW;
+
+import java.util.concurrent.ExecutorService;
+
+import org.junit.ClassRule;
+import org.junit.rules.TestRule;
+
+import io.vertx.core.spi.metrics.MetricsContextTest;
+
+/**
+ * This test class checks the Metrics functionality of Vert.x is not impacted by
+ * using a service loaded
+ * {@link io.vertx.core.spi.executor.ExecutorServiceFactory} to create an
+ * {@link ExecutorService} that is then used to run worker tasks.
+ * 
+ * It performs the same tests as MetricsContextTest but using an external
+ * {@link java.util.concurrent.ExecutorService} for worker tasks
+ * 
+ * To enable different tests to have particular ExecutorServiceFactory's (or
+ * none), we only put the particular ExecutorServiceFactory .class file on the
+ * classpath and make it service loadable as part of each test and are careful
+ * to remove it afterwards.
+ */
+public class ExecutorFactoryMetricsTest extends MetricsContextTest {
```

Comment:
I don't think we need this as it tests metrics behavior.

Reply:
Agreed.

Reply:
Done.